js/plugins/iq.js

```diff
@@ -0,0 +1,348 @@
+        this.ooyalaReporter.reportReplay();
+        OO.log("IQ: Reported: reportReplay()");
+        break;
+
+      case OO.Analytics.EVENTS.AD_REQUEST:
+      case OO.Analytics.EVENTS.AD_REQUEST_SUCCESS:
+      case OO.Analytics.EVENTS.AD_SDK_LOADED:
+      case OO.Analytics.EVENTS.AD_SDK_LOAD_FAILURE:
+      case OO.Analytics.EVENTS.AD_BREAK_STARTED:
+      case OO.Analytics.EVENTS.AD_BREAK_ENDED:
+      case OO.Analytics.EVENTS.AD_POD_STARTED:
+      case OO.Analytics.EVENTS.AD_POD_ENDED:
+      case OO.Analytics.EVENTS.AD_STARTED:
+      case OO.Analytics.EVENTS.AD_ENDED:
+      case OO.Analytics.EVENTS.AD_SKIPPED:
+      case OO.Analytics.EVENTS.AD_ERROR:
+      case OO.Analytics.EVENTS.AD_REQUEST_EMPTY:
+      case OO.Analytics.EVENTS.AD_REQUEST_ERROR:
+      case OO.Analytics.EVENTS.AD_PLAYBACK_ERROR:
+      case OO.Analytics.EVENTS.AD_IMPRESSION:
+      case OO.Analytics.EVENTS.AD_SDK_IMPRESSION:
+      case OO.Analytics.EVENTS.AD_COMPLETED:
+      case OO.Analytics.EVENTS.AD_CLICKTHROUGH_OPENED:
+      case OO.Analytics.EVENTS.SDK_AD_EVENT:
+        var eventMetadata = params[0];
+        if(!eventMetadata)
+        {
+          eventMetadata = {};
+        }
+
+        if (eventMetadata.adEventName)
+        {
+          eventMetadata.adEventName = eventName + ":" + eventMetadata.adEventName;
+        }
+        else
+        {
+          eventMetadata.adEventName = eventName;
+        }
+        this.ooyalaReporter.reportCustomEvent(eventName, eventMetadata);
+        OO.log("IQ: Reported: reportCustomEvent() for event: " + eventName + " with args:" + JSON.stringify(eventMetadata));
+        break;
+      default:
+        break;
+    }
+  };
+
+  /**
+   * [Required Function] Clean up this plugin so the garbage collector can clear it out.
+   * @public
+   * @method IqPlugin#destroy
+   */
+  this.destroy = function ()
+  {
+    _framework = null;
+  };
+
+  /**
+   * Called when the SDK fails to load.
+   * @private
+   * @method NielsenAnalyticsPlugin#sdkLoadError
+   */
+  var sdkLoadError = function()
+  {
+    //Destroy and unregister
+    if (_.isString(id))
+    {
+      framework.unregisterPlugin(id);
+    }
+    this.destroy();
+  };
+
+  var trySetupAnalytics = OO._.bind(function()
+  {
+    if (window.Ooyala)
+    {
+      this.ooyalaReporter = new Ooyala.Analytics.Reporter();
+
+      var missedEvents;
+      //if you need to process missed events, here is an example
+      if (_framework && OO._.isFunction(_framework.getRecordedEvents))
+      {
+        missedEvents = _framework.getRecordedEvents();
+        for (var i = 0; i < missedEvents.length; i++)
+        {
+          recordedEvent = missedEvents[i];
+          this.processEvent(recordedEvent.eventName, recordedEvent.params);
+        }
+      }
+
+      // TODO: setup
+      var deviceInfo = {};
+      var playerName = "Ooyala Player";
+      var playerVersion = "v4"; // TODO: need a mechanism in core to get this
```

Comment:
Try OO.VERSION.core.releaseVersion.  Hopefully you have access to it from the plugin.  Otherwise we may need to expose it.  I got it from this section of code that is dynamically filled in during the build process:

try {
(function() {
  var OO = {};

  OO.VERSION = {
    "core" : {"releaseVersion": "4.14.9", "rev": "<CORE_REV>"}
  };

  OO.playerParams = {
    "core_version" : 4,
    "vast_proxy_url" : "http://player.ooyala.com/adinsertion/vast_proxy"
  };

Reply:
Done, it was in scope